packages/next/src/client/components/layout-router.tsx

```diff
@@ -263,6 +262,9 @@ class InnerScrollAndFocusHandler extends React.Component<ScrollAndFocusHandlerPr
         }
       )
 
+      // State is mutated to ensure that the focus and scroll is applied only once.
```

Comment:
Just a small comment nit!

```suggestion
      // Mutate after scrolling so that it can be read by `handleSmoothScroll`
```

Reply:
Appreciate it! This makes more sense. I just copied it from above.